src/pages/Search/index.js

```diff
@@ -1,25 +1,27 @@
-import React from 'react';
-import dayjs from 'dayjs';
-import axios from 'axios';
-import SearchContainer from './styles';
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import loadData from '../../api/posts';
+import Loader from '../../components/loader';
 import Form from './components/Form';
-import config from '../../constants/config';
+import SearchContainer from './styles';
 
 const SearchPage = () => {
-  const loadData = async (subreddit) => {
-    const size = 500;
-    const date = dayjs()
-      .subtract(1, 'year')
-      .unix();
+  const [loading, setLoading] = useState(false);
+  const { subreddit } = useParams();
 
-    const query = `?subreddit=${subreddit}&size=${size}&sort=desc&sort_type=score&after=${date}`;
-    const URL = `${config.BASE_URL}search/submission/${query}`;
-    await axios.get(URL);
-  };
+  useEffect(() => {
+    setLoading(true);
+    const onLoad = async () => {
+      await loadData(subreddit);
```

Comment:
The try..catch block is still missing here for error handling. You can also do that in another branch. I'll leave that up to you